### ISSUE TITLE
Comparison to uninitialized value in test

### ISSUE THREAD

**Issue by danluu** [stan-dev/math#819]
#### Description:

This is very minor, but `check_greater_test.cpp` has:
 low = std::numeric_limits<double>::infinity();
  EXPECT_THROW(check_greater(function, "x", x, low), std::domain_error)
      << "check_greater: matrix<3, 1>, double, should fail with infinity";

where `x` has been declared but not initialized. This causes checking tools to complain and this seems worth cleaning up because it's nice to be clean on "use without initializing" warnings since those can sometimes indicate serious errors.

I'll probably try to fix this in a PR that aggregates a bunch of cleanup stuff, something like #813.

#### Current Version:
v2.17.0

**Comment by bob-carpenter** [comment]
Yes, this should definitely be fixed.  We should try to get rid of everywhere we depend on undefined behavior.
